model/config.cf

```diff
@@ -53,15 +54,20 @@ entity Block:
         the different element of the list.  (Siblings of this element)
         If it is a dict, the key attribute should be set, and will be used to place
         the element in the dict named after the name of this block.
-    :attr key: The key, required for list and dict nesting mode, ignored otherwise.
+    :attr key: The key, required for list and dict nesting mode, automatically set otherwise.
     :attr _config: Generated, the serialized version of this config.
+    :attr _state: Generated, the current state for the resource attached to this config
```

Comment:
I don't understand why this is here and what form it is expected to have (if the backend system generating this is very flexible in version, how is this kept stable here?)

Reply:
The state dict here will always have the same form, this is the dict returned by the terraform provider.  The idea here is to attach to each config block, the corresponding state dict, so that any part of the state can be accessed easily (no need to use complex dict_path for example).

The config block then has two main functions:
1. Build the config, from the leaves to the root, it will aggregate everything into a dict (this is stored in the `_dict` attribute)
2. Build the state, from the root to the leaves, it will extract from the parent state, the part corresponding to the current block, and this is stored in the `_state` attribute.

Reply:
This might illustrate better the intent between this `_state` dict:
```
implementation config for vsphere::VsphereVnic:
    """
    Build the config block corresponding to this entity and attaching all the non-computed entity attributes to it.

    """
    # Building the config block for this entity
    self._config_block = terraform::config::Block(
        name=null,
        attributes={
            "distributed_port_group": self.distributed_port_group,
            "distributed_switch_port": self.distributed_switch_port,
            "host": self.host,
            "netstack": self.netstack,
            "portgroup": self.portgroup,
        },
        deprecated=false,
        nesting_mode="single",
    )
end

implementation state for vsphere::VsphereVnic:
    """
    Attach all the generated attributes values to the entity.

    """
    self.id = self._config_block._state["id"]
    self.mac = self._config_block._state["mac"]
    self.mtu = self._config_block._state["mtu"]
end
```